sourmash-protein/Snakefile

```diff
@@ -0,0 +1,81 @@
+# load stuff from config
+SAMPLES=config.get('metagenome_files', [])
+QUERY=config.get('query_file', '')
+ksize = config.get('ksize', 'unset')
+
+try:
+    ksize = int(ksize)
+except ValueError:
+    print(f"Invalid ksize {repr(ksize)}", file=sys.stderr)
+    raise
+
+# this sets LOG_DIR, RESULTS_DIR, SAMPLE_PREP, and QUERY_PREP
+include: '../common.snakefile'
+
+###
+### actual rules to run something
+###
+
+rule prepare_sample:
+    output: directory(SAMPLE_PREP)
+    conda: "../conf/env/sourmash4.yml"
+    log: f"{LOG_DIR}/prepare_sample.log"
+    benchmark: f"{LOG_DIR}/prepare_sample.benchmark"
+    shell: """
+       rm -fr {output}
+       mkdir -p {output}
+       sourmash sketch translate {SAMPLES} -o {output}/sample.sig 2> {log}
+    """
+
+rule prepare_query:
+    output: directory(QUERY_PREP)
+    conda: "../conf/env/sourmash4.yml"
+    params:
+        name = os.path.basename(QUERY)
+    log: f"{LOG_DIR}/prepare_query.log"
+    benchmark: f"{LOG_DIR}/prepare_query.benchmark"
+    shell: """
+       rm -fr {output}
+       mkdir -p {output}
+       sourmash sketch translate {QUERY} -o {output}/query.sig 2> {log}
```

Comment:
please add parameter string here, e.g. `-p`, as in `sourmash/Snakefile` - thanks!

Reply:
done